### DISCUSSION TITLE
Search results in the web application and API returns different results

### DISCUSSION THREAD

**Discussion by shusterboris** [oracle/opengrok#4506] [Q&A]
Dear friends!\
I faced the problem that I meant in the subject. 
The problem is that the search returns a different number of results depending on the source of the request: UI or API. For example
UI query is..
/search?project=q11_integ&full=odui&defs=&refs=&path=&type=&xrd=&nn=1&searchall=true
result is...
![image](https://github.com/oracle/opengrok/assets/25655449/34bc547d-e924-4b87-811b-78f7445882fc)
API request (using curl):  `curl -k -o res.txt -v "http://localhost:8080/api/v1/search?projects=q11_integ&full=odui&maxresults=200000" `
Result is: 
```json
{"time":70220,"resultCount":37504,"startDocument":0,"endDocument":**37500**,"results":{"
```
I don't know if this matters, I observe this effect for large samples

**Comment by shusterboris** [comment]
Has anyone encountered such a problem?

**Reply by vladak** [reply]
There is https://github.com/oracle/opengrok/issues/3239 - except in your case it is the other way round w.r.t. result count.

Sufficient debug should be added to see what the low level query difference is.

**Reply by vladak** [reply]
Does this also happen right after the webapp is redeployed/appserver restarted ? I am asking to see if this could be caused by stale `IndexSearcher` object perhaps.

**Comment by shusterboris** [comment]
Hi, Vladimir. Thank you for response.
Unfortunately, not. I checked it.

Respectfully, Boris

вт, 19 дек. 2023 г., 12:29 Vladimir Kotal ***@***.***>:
